dev-lang/zig/zig-0.13.0-r1.ebuild

```diff
@@ -0,0 +1,242 @@
+
+	SRC_URI="
+		https://ziglang.org/download/${PV}/${P}.tar.xz
+		verify-sig? ( https://ziglang.org/download/${PV}/${P}.tar.xz.minisig )
+	"
+	KEYWORDS="~amd64 ~arm ~arm64"
+
+	BDEPEND="verify-sig? ( sec-keys/minisig-keys-zig-software-foundation )"
+fi
+
+# project itself: MIT
+# There are bunch of projects under "lib/" folder that are needed for cross-compilation.
+# Files that are unnecessary for cross-compilation are removed by upstream
+# and therefore their licenses (if any special) are not included.
+# lib/libunwind: Apache-2.0-with-LLVM-exceptions || ( UoI-NCSA MIT )
+# lib/libcxxabi: Apache-2.0-with-LLVM-exceptions || ( UoI-NCSA MIT )
+# lib/libcxx: Apache-2.0-with-LLVM-exceptions || ( UoI-NCSA MIT )
+# lib/libc/wasi: || ( Apache-2.0-with-LLVM-exceptions Apache-2.0 MIT BSD-2 ) public-domain
+# lib/libc/musl: MIT BSD-2
+# lib/libc/mingw: ZPL public-domain BSD-2 ISC HPND
+# lib/libc/glibc: BSD HPND ISC inner-net LGPL-2.1+
+LICENSE="MIT Apache-2.0-with-LLVM-exceptions || ( UoI-NCSA MIT ) || ( Apache-2.0-with-LLVM-exceptions Apache-2.0 MIT BSD-2 ) public-domain BSD-2 ZPL ISC HPND BSD inner-net LGPL-2.1+"
+SLOT="${ZIG_SLOT}"
+IUSE="doc +llvm test"
+RESTRICT="!test? ( test )"
+REQUIRED_USE="
+	!llvm? ( !doc )
+	llvm? ( ${LLVM_REQUIRED_USE} )
+"
+
+# Used by both cmake and zig-build eclass.
+BUILD_DIR="${WORKDIR}/${P}_build"
+
+# Zig requires zstd and zlib compression support in LLVM, if using LLVM backend.
+# (non-LLVM backends don't require these)
+# They are not required "on their own", so please don't add them here.
+# You can check https://github.com/ziglang/zig-bootstrap in future, to see
+# options that are passed to LLVM CMake building (excluding "static" ofc).
+DEPEND="
+	llvm? (
+		$(llvm_gen_dep '
+			sys-devel/clang:${LLVM_SLOT}
+			sys-devel/lld:${LLVM_SLOT}
+			sys-devel/llvm:${LLVM_SLOT}[zstd]
+		')
+	)
+"
+BDEPEND+="
+	test? ( !!<sys-apps/sandbox-2.39 )
+	${DEPEND}
+"
+RDEPEND="${DEPEND}"
+IDEPEND="app-eselect/eselect-zig"
+
+DOCS=("README.md" "doc/build.zig.zon.md")
+
+PATCHES=(
+	"${FILESDIR}/zig-0.13.0-test-std-kernel-version.patch"
+	"${FILESDIR}/zig-0.13.0-skip-test-stack_iterator.patch"
+)
+
+# zig-build does not set this for us since we use ZIG_OPTIONAL=1
+QA_FLAGS_IGNORED="usr/.*/zig/${PV}/bin/zig"
+
+# Since commit https://github.com/ziglang/zig/commit/e7d28344fa3ee81d6ad7ca5ce1f83d50d8502118
+# Zig uses self-hosted compiler only
+CHECKREQS_MEMORY="4G"
+
+pkg_setup() {
+	# Skip detecting zig executable.
+	ZIG_EXE="not-applicable" ZIG_VER="${PV}" zig-build_pkg_setup
+
+	export ZIG_SYS_INSTALL_DEST="${EPREFIX}/usr/$(get_libdir)/zig/${PV}"
+
+	if use llvm; then
+		tc-is-cross-compiler && die "USE=llvm is not yet supported when cross-compiling"
+		llvm-r1_pkg_setup
+	fi
+
+	check-reqs_pkg_setup
+}
+
+src_unpack() {
+	if [[ ${PV} == 9999 ]]; then
+		git-r3_src_unpack
+	else
+		verify-sig_src_unpack
+	fi
+}
+
+src_prepare() {
+	# Avoid double patching. Copied from net-misc/sunshine-0.23.1::gentoo .
+	default_src_prepare
```

Comment:
Please don't do this. I don't think this is well-defined.

Reply:
I can remove zig-build_src_prepare call here since it now has much less logic, and system mode is not needed when bootstrapping here since they do not use remote dependencies. Is this fine?

```bash
if use llvm; then
    cmake_src_prepare
else
    # Sync with zig-build_src_prepare
    default_src_prepare
    mkdir -p "${BUILD_DIR}" || die
fi
```

Reply:
Yes, thanks!

Reply:
0fa14e036947d60c2afe4cac3c1c50669bb3722b